### PR TITLE
Enable virtual-host-style requests via MINIO_DOMAIN.

### PR DESCRIPTION
**Pull request by iamazeem**
Enable virtual-host-style requests via [MINIO_DOMAIN](https://github.com/minio/minio/blob/master/docs/config/README.md#domain).

Signed-off-by: Azeem Sajid <azeem.sajid@gmail.com>

### CODE REVIEW

**Review by OriHoch — CHANGES_REQUESTED** [review]
looks good, please add a section in the README for testing this feature locally with docker-compose

File: nginx/minio_proxy_pass.conf [hunk 1]
```diff
@@ -1 +1 @@
-proxy_pass http://minio:8080;
+proxy_pass http://minio-server:8080;
```
Comment (OriHoch) [inline nginx/minio_proxy_pass.conf:1]: @iamazeem why did you change this? this is the cause for CI error

Comment (OriHoch) [reply nginx/minio_proxy_pass.conf:1]: https://github.com/CloudWebManage/cwm-worker-deployment-minio/runs/3333987304?check_suite_focus=true#step:3:2028

Comment (iamazeem) [reply nginx/minio_proxy_pass.conf:1]: The nginx pod was crashing due to this.
The `minio-server` was previously used here: https://github.com/CloudWebManage/cwm-worker-deployment-minio/blob/ecf6410e7ba22b8141c58181c0dd3814b2ff16a9/helm/templates/nginx-deployment.yaml#L43-L44

Comment (OriHoch) [reply nginx/minio_proxy_pass.conf:1]: yes, for the helm deployment MINIO_PROXY_PASS_HOST is set to minio-server
for docker-compose it's not set and remains with the defailt file

Comment (iamazeem) [reply nginx/minio_proxy_pass.conf:1]: Need to revert it then for `docker-compose` if it should remain the same?
So, for helm, `.values.yaml` can be used to configure `MINIO_PROXY_PASS_HOST`, right?

Comment (OriHoch) [reply nginx/minio_proxy_pass.conf:1]: No change is needed, I just tested it and helm deployment works. If you had some error it needs to be debugged
the MINIO_PROXY_PASS_HOST should not be changed

Comment (iamazeem) [reply nginx/minio_proxy_pass.conf]: Reverted. Tested with `docker-compose` and it worked:

Bucket: test
Object: file.txt

/etc/hosts:

```text
# ...
127.0.0.1 example001.com test.example001.com
```

curl:

```shell
$ curl 'http://example001.com:8080/test/file.txt'
axbycz

$ curl 'http://test.example001.com:8080/file.txt'
axbycz
```

nginx logs:

```text
...
nginx_1              | 172.19.0.1 - - [16/Aug/2021:16:17:19 +0000] "GET /test/file.txt HTTP/1.1" 200 7 "-" "curl/7.58.0" "-"
...
nginx_1              | 172.19.0.1 - - [16/Aug/2021:16:17:33 +0000] "GET /file.txt HTTP/1.1" 200 7 "-" "curl/7.58.0" "-"
```

Comment (iamazeem) [reply nginx/minio_proxy_pass.conf]: The helm deployment is also working. :thinking: 
Tried to reproduce a couple of times but could not.
I might have had something in the environment at that point in time.

**Comment by OriHoch** [comment]
@iamazeem looks good using the default domain, but I think additional support is needed for cases where the hostname is in additional domains and using SSL:

add to hosts file:

```
127.0.0.0 test.example002.com test.example003.com
```

make https request and check SSL headers:

* `curl -vk https://test.example002.com:8443/file.txt`
* `curl -vk https://test.example003.com:8443/file.txt`

expected curl response headers to contain:

* for request to test.example002.com: `CN=example002.com`
* for request to test.example003.com: `CN=example003.com`

I think changes will be needed in nginx to support hostname *.DOMAIN or something like that..

**Comment by iamazeem** [comment]
@OriHoch : Here are the CN values:

**example002.com**:

```shell
$ curl -vk https://example002.com:8443/test/file.txt    # CN=example002.com
$ curl -vk https://test.example002.com:8443/file.txt    # CN=example002.com
```

**example003.com**:

```shell
$ curl -vk https://example003.com:8443/test/file.txt    # CN=example003.com
$ curl -vk https://test.example003.com:8443/file.txt    # CN=example002.com
```

**Discrepancy**:

```text
test.example003.com > CN=example002.com
```

**Comment by iamazeem** [comment]
For SSL, with `*.domain`:

```text
sed "s/__SERVER_NAME__/server_name ${host_name} *.${host_name};/" "$DEFAULT_HTTPS_CONF" > "$https_conf"
```

It's working fine now.

**example002.com**:

```shell
$ curl -vk https://example002.com:8443/test/file.txt    # CN=example002.com
$ curl -vk https://test.example002.com:8443/file.txt    # CN=example002.com
```

**example003.com**:

```shell
$ curl -vk https://example003.com:8443/test/file.txt    # CN=example003.com
$ curl -vk https://test.example003.com:8443/file.txt    # CN=example003.com
```

**Review by OriHoch — APPROVED** [review]

**Pull request merged**